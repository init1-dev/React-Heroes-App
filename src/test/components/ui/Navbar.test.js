import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";

import { AuthContext } from "../../../auth/authContext";
import { Navbar } from "../../../components/ui/Navbar";
import { types } from "../../../types/types";

const mockNavigate = jest.fn();
const logout = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('pruebas en <NavBar />', () => {

    const contextValue = {
        user: {
            logged: true,
            name: 'Pedro'
        },
        dispatch: (type) => logout(type)
    };
    
    test('debe mostrarse correctamente', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={ ['/'] }>
                    <Navbar />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text-info').text().trim() ).toBe('Pedro');
    });
    
    test('debe llamar el logout, llamar el navigate y el dispatch con los argumentos', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={ ['/'] }>
                    <Navbar />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        wrapper.find('.btn').simulate('click');

        expect( mockNavigate ).toHaveBeenCalledWith('login', {
            replace: true
        });

        expect( logout ).toHaveBeenLastCalledWith({ type: types.logout });
    });
    
    
});
