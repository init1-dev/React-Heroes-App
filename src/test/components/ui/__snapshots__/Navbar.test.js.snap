// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <NavBar /> debe mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand ms-3"
    href="/"
    onClick={[Function]}
  >
    React-Heroes-App
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        className="nav-item nav-link "
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav-item nav-link "
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav-item nav-link "
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
  >
    <ul
      className="navbar-nav ml-auto me-3"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Pedro
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
